src/components/views/elements/MemberEventListSummary.js

```diff
@@ -140,15 +145,31 @@ module.exports = React.createClass({
         // Reorder events so that joins come before leaves
         let eventsToRender = this.props.events;
 
-        // Filter out those who joined, then left
-        let filteredEvents = eventsToRender.filter(
-            (e) => {
-                return eventsToRender.filter(
-                    (e2) => {
-                        return e.getSender() === e2.getSender()
-                            && e.event.content.membership !== e2.event.content.membership;
-                    }
-                ).length === 0;
+        // Create an array of events that are not "cancelled-out" by another
+        // A join of sender S is cancelled out by a leave of sender S etc.
+        let filteredEvents = [];
+                        // return e.getSender() === e2.getSender()
+                        //     && e.event.content.membership !== e2.event.content.membership;
+        let senders = new Set(eventsToRender.map((e) => e.getSender()));
+        senders.forEach(
+            (userId) => {
+                // Only push the last event if it isn't cancelled out
+
+                let userEvents = eventsToRender.filter((e) => {
+                    return e.getSender() === userId;
+                }).sort((a, b) => {
```

Comment:
It's just so that the summary always appears in the same order; only the first few users are mentioned in the summary and I decided that 'first' might as well be based on the event timestamps.